internal/git/git.go

```diff
@@ -417,8 +419,14 @@ func (g *Git) GetDownloadLink(version string) (string, string, error) {
 
 	for _, release := range releases {
 		for _, asset := range release.Assets {
-			if strings.Contains(strings.ToLower(asset.GetName()), "_linux_x86_64") || strings.Contains(strings.ToLower(asset.GetName()), "_linux_amd64") {
-				if version == "latest" || version == release.GetTagName() {
+			if version == "latest" || version == release.GetTagName() {
+				var utsname unix.Utsname
+				err = unix.Uname(&utsname)
+				if err != nil {
+					return "", "", fmt.Errorf("failed to get uname: %w", err)
+				}
+
+				if strings.Contains(strings.ToLower(asset.GetName()), strings.ToLower(runtime.GOOS)) && strings.Contains(strings.ToLower(asset.GetName()), strings.ToLower(strings.Trim(string(utsname.Machine[:]), "\x00"))) {
```

Comment:
was this to support running on mac? @ryan-timothy-albert this might be what you are running into if you are on mac as it was always downloading the linux binary before

Reply:
@ThomasRooney does this always match the release names in the speakeasy repo?

previously it was matching against them in two different formats to support backwards compatible pinning to older versions, I have a feeling this will break that now

Reply:
well `x86_64, amd64` are the same thing I would only expect it to return one and likely `amd64` and older versions of the CLI don't have `amd64` artifacts and new versions don`t have `x86_64` artifacts

Reply:
well `x86_64, amd64` are the same thing I would only expect it to return one and likely `amd64` and older versions of the CLI don't have `amd64` artifacts and new versions don`t have `x86_64` artifacts

Reply:
Hmm; that's fair. I'll add back in the older condition for safety. My understanding is that `uname.Machine` does differ from arch in some scenarios: https://en.wikipedia.org/wiki/Uname. Presume you're happy with `runtime.GOARCH` instead?

Reply:
yeah I believe that is what goreleaser uses which is where this change came from. But yeah we still do need to make sure we match against `_x86_64` when `_amd64` doesn't exist

Reply:
Added a special condition for `linux` so we don't break that, and referenced this conversation.